superset-frontend/plugins/plugin-chart-echarts/src/Graph/transformProps.ts

```diff
@@ -242,14 +248,39 @@ export default function transformProps(
     }
     const sourceName = link[source] as string;
     const targetName = link[target] as string;
-    const sourceCategoryName = sourceCategory
-      ? getCategoryName(sourceCategory, link[sourceCategory])
-      : undefined;
-    const targetCategoryName = targetCategory
-      ? getCategoryName(targetCategory, link[targetCategory])
-      : undefined;
-    const sourceNode = getOrCreateNode(sourceName, source, sourceCategoryName);
-    const targetNode = getOrCreateNode(targetName, target, targetCategoryName);
+    let sourceCategoryName;
+    let targetCategoryName;
+    let sourceNodeColor = colorFn('node');
+    let targetNodeColor = colorFn('node');
+    const linkColor = colorFn('link');
+
+    if (sourceCategory) {
+      sourceCategoryName = getCategoryName(
+        sourceCategory,
+        link[sourceCategory],
+      );
+      sourceNodeColor = colorFn(sourceCategoryName);
+    }
+    if (targetCategory) {
+      targetCategoryName = getCategoryName(
+        targetCategory,
+        link[targetCategory],
+      );
+      targetNodeColor = colorFn(targetCategoryName);
+    }
```

Comment:
Can this let assignment consolidate the logic like this?
 
```suggestion
    const sourceCategoryName = sourceCategory ? getCategoryName(
        sourceCategory,
        link[sourceCategory],
      ) : undefined;
    const  targetCategoryName = targetCategory ? getCategoryName(
        targetCategory,
        link[targetCategory],
      ) : undefined;
    const sourceNodeColor = sourceCategory ? colorFn(sourceCategoryName) : colorFn('node');
    const targetNodeColor = targetCategory ? colorFn(targetCategoryName) : colorFn('node');
    const linkColor = colorFn('link');
```